include/boost/circular_buffer/base.hpp

```diff
@@ -1515,7 +1518,8 @@ private empty_value<Alloc>
                 if (empty())
                     return;
                 decrement(m_first);
-                replace(m_first, value_type(::boost::forward<V>(value)));
+                destroy_item(m_first);
+                boost::allocator_construct(alloc(), boost::to_address(m_first), ::boost::forward<Args>(args)...);
```

Comment:
This has not been compiled with BOOST_NO_CXX11_VARIADIC_TEMPLATES defined because I do not think this would compile.

There is no variadic template parameter Args (args) here